src/Tabs/TabsList.js

```diff
@@ -1,15 +1,74 @@
 /* eslint-disable jsx-a11y/no-noninteractive-element-interactions */
-import React, { useEffect, useContext } from "react";
+import debounce from "lodash.debounce";
 import PropTypes from "prop-types";
+import React, { useContext, useEffect, useState } from "react";
+import Arrow from "./Arrow";
 import TabsContext from "./context";
 
 const noop = () => {};
 
 const TabsList = ({ children, xPadding = "none" }) => {
-  const { tabIds, setTabIds, changeTabs, currentIndex, hasPanels } =
-    useContext(TabsContext);
+  const [showLeftArrow, setShowLeftArrow] = useState(false);
+  const [showRightArrow, setShowRightArrow] = useState(false);
+
+  const {
+    tabIds,
+    setTabIds,
+    changeTabs,
+    currentIndex,
+    hasPanels,
+    tabsListRef,
+    setIsResponsive,
+  } = useContext(TabsContext);
   const childArray = React.Children.toArray(children);
 
+  const updateScrollButtonState = () => {
+    const { scrollWidth, clientWidth, scrollLeft } = tabsListRef.current;
+    const nextShowLeftArrow = scrollLeft > 1;
+    const nextShowRightArrow = scrollLeft < scrollWidth - clientWidth - 1;
+
+    setShowLeftArrow(nextShowLeftArrow);
+    setShowRightArrow(nextShowRightArrow);
+    setIsResponsive(nextShowLeftArrow || nextShowRightArrow);
+  };
+
+  useEffect(() => {
+    if (tabsListRef.current) {
+      updateScrollButtonState();
+    }
+  }, []);
+
+  useEffect(() => {
+    const handleScrollButtonsUpdate = debounce(() => {
```

Comment:
Good idea to debounce these calls. Another option is to use `raf-schd` (already included in NDS) to run the operation based on the next animation frame instead of on a 100ms debounce.

Example: https://github.com/narmi/design_system/blob/48d2b2f3a881a718837acc94ec5a2d24308b941f/src/Dialog/index.js#L50